modules/340-monitoring-custom/templates/service-monitor.yaml

```diff
@@ -17,13 +17,24 @@ spec:
 
   # HTTPS target with samples limited to 1000
   - scheme: https
-    {{- include "tls_config" . | nindent 4 }}
+    {{- include "tls_config" "" | nindent 4 }}
     relabelings:
     {{- include "service_label_selector" . | nindent 4 }}
     {{- include "keep_targets_for_schema" (list "service" "https") | nindent 4 }}
     {{- include "endpoint_by_service_port_name" "https" | nindent 4 }}
     {{- include "base_relabeling" "service" | nindent 4 }}
 
+    {{- if .Values.monitoringCustom.internal.prometheusScraperIstioMTLSEnabled }}
```

Comment:
I'd like to avoid additional the secret subscription hook and this if clause. In my opinion it is ok to always have this monitor part if istio is enabled in the cluster.

Reply:
No, if there istn't a mtls-client-secret, prometheus swears dirty.

Reply:
Yeah, but if mtls is enabled it will be restarted monthly?

Reply:
Dirty in which way?

Reply:
I have an idea to remove the annotation with mtls secret checksum from the Prometheus resource and add an annotation checksum for PodMonitor and ServiceMonitor, so when the certificate changes, only these two resources will be recreated, and Prometheus will not restart.

Reply:
Prometheus skips ServiceMonitor (or PodMonitor) if there isn't a secret.